### PR TITLE
Mbasiuk/tdi 41415 netsuite retry mechanism

### PR DESCRIPTION
**Pull request by mbasiuk-talend**
**What is the current behavior?** (You can also link to an open issue here)
Retry mechanism uses static value of 5 seconds for retries.

https://jira.talendforge.org/browse/TDI-41415

**What is the new behavior?**
Improve logic for retrying using exponential parameter with multiplier x2. This should save parallel execution from being locked by expensive operation on NetSuite side that might lasts more than 7 seconds.

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [x] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [x] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: components/components-netsuite/components-netsuite-runtime/src/test/java/org/talend/components/netsuite/test/client/TestNetSuiteClientService.java [hunk 1]
```diff
@@ -177,7 +177,7 @@ protected void doLogin() throws NetSuiteException {
             }
 
             if (i != getRetryCount() - 1) {
-                waitForRetryInterval();
+                waitForRetryInterval(i);
```
Comment (undx) [inline components/components-netsuite/components-netsuite-runtime/src/test/java/org/talend/components/netsuite/test/client/TestNetSuiteClientService.java]: At first sight, the classes in this package look like production classes... :thinking:

File: components/components-netsuite/components-netsuite-runtime/src/main/java/org/talend/components/netsuite/client/NetSuiteClientService.java [hunk 2]
```diff
@@ -54,6 +54,11 @@
 
     public static final int DEFAULT_SEARCH_PAGE_SIZE = 100;
 
+    /** Base Interval between retries. */
+    public static final int DEFAULT_RETRY_INTERVAL = 2;
```
Comment (undx) [inline components/components-netsuite/components-netsuite-runtime/src/main/java/org/talend/components/netsuite/client/NetSuiteClientService.java]: As it's a fixed value that cannot be changed, maybe call be also called `FIXED`? ;-)
So, the behavior is not customizable?

File: components/components-netsuite/components-netsuite-runtime/src/main/java/org/talend/components/netsuite/client/NetSuiteClientService.java [hunk 3]
```diff
@@ -54,6 +54,11 @@
...
+
+    private static final int FIXED_RETRY_TIMEOUT = 5;
```
Comment (undx) [inline components/components-netsuite/components-netsuite-runtime/src/main/java/org/talend/components/netsuite/client/NetSuiteClientService.java]: maybe should be called something like minimal wait time interval?

File: components/components-netsuite/components-netsuite-runtime/src/main/java/org/talend/components/netsuite/client/NetSuiteClientService.java (hunk 2 above)
Comment (mbasiuk-talend) [reply components/components-netsuite/components-netsuite-runtime/src/main/java/org/talend/components/netsuite/client/NetSuiteClientService.java]: Agree.

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/69/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/69/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Ambasiuk%2FTDI-41415-netsuite_retry_mechanism)

**Pull request merged**